### PR TITLE
[SPARK-18346][SQL] TRUNCATE TABLE should fail if no partition is matched for the given non-partial partition spec

### PR DESCRIPTION
**Pull request by cloud-fan**
## What changes were proposed in this pull request?

a follow up of https://github.com/apache/spark/pull/15688

## How was this patch tested?

updated test in `DDLSuite`

### CODE REVIEW

**Comment by cloud-fan** [comment]
cc @gatorsmile

**Comment by rxin** [comment]
LGTM otherwise.

File: sql/core/src/main/scala/org/apache/spark/sql/execution/command/tables.scala [hunk 1]
```diff
@@ -324,38 +324,45 @@ case class TruncateTableCommand(
...
         val normalizedSpec = partitionSpec.map { spec =>
           PartitioningUtils.normalizePartitionSpec(
             spec,
-            table.partitionColumnNames,
+            partCols,
             table.identifier.quotedString,
             spark.sessionState.conf.resolver)
         }
-        catalog.listPartitions(table.identifier, normalizedSpec).map(_.storage.locationUri)
+        val parts =
```
Comment (gatorsmile) [inline sql/core/src/main/scala/org/apache/spark/sql/execution/command/tables.scala]: Nit: How about `partLocations`?

**Comment by gatorsmile** [comment]
LGTM except a minor comment and test pending Jenkins

**Comment by rxin** [comment]
LGTM pending Jenkins.

**Comment by cloud-fan** [comment]
thanks for the review, merging to master!

**Pull request closed without merging**